sqlmesh/core/engine_adapter/spark.py

```diff
@@ -38,14 +44,53 @@ def spark(self) -> PySparkSession:
     def _use_spark_session(self) -> bool:
         return True
 
-    def _ensure_pyspark_df(self, generic_df: DF) -> PySparkDataFrame:
+    @classmethod
+    def convert_columns_to_types_to_pyspark_schema(
+        cls, columns_to_types: t.Dict[str, exp.DataType]
+    ) -> t.Optional[spark_types.StructType]:
+        from pyspark.sql import types as spark_types
+
+        mapping = {
+            exp.DataType.Type.BOOLEAN: spark_types.BooleanType,
+            exp.DataType.Type.TEXT: spark_types.StringType,
+            exp.DataType.Type.INT: spark_types.IntegerType,
+            exp.DataType.Type.BIGINT: spark_types.LongType,
+            exp.DataType.Type.FLOAT: spark_types.FloatType,
+            exp.DataType.Type.DOUBLE: spark_types.DoubleType,
+            exp.DataType.Type.DECIMAL: spark_types.DecimalType,
+            exp.DataType.Type.DATE: spark_types.DateType,
+            exp.DataType.Type.TIMESTAMP: spark_types.TimestampType,
+        }
+        try:
```

Comment:
why are complex types not supported? just because it's harder to write the code? i don't think it's that much more complicated to support maps and arrays etc

Reply:
Yes just saving time. I don't have much before I'm OOO and I want to make sure SQLMesh works for Databricks users by getting this essential fixes into main first.

Reply:
ok, put it on your todo to fix this properly